app/stock/__init__.py

```diff
@@ -1,10 +1,61 @@
 import robin_stocks
+import datetime
+import pytz
+import time
+import os
 
 from config import settings
 from app.stock import infomation
 
 login = robin_stocks.login(settings.USER_EMAIL, settings.USER_PASSWORD)
 
+_LAST_FILE_HOUR_ = None
+_LOG_FILE_NAME_ = None
+
 
 def run_service():
-    infomation.build_holdings()
+    while True:
+        holdings = None
+        market_info = infomation.market_open_time(settings.MARKETS[0])
+        if not market_info['is_open_today']:
+            logger("sleep during holiday")
+            time.sleep(60 * 60)
+        elif market_info['is_open_now']:
+            logger("work during open hours")
+            holdings = infomation.build_holdings()
+        elif market_info['end - utcnow'] <= 0:
+            logger("sleep after market closed")
+            time.sleep(60 * 60)
+        elif market_info['utcnow - start'] > -60 * 5:
```

Comment:
line 29, 34, should be market_info['utcnow - open']

Reply:
I'll use `utcnow - start` for all, as `open` is a python keyword.